protected/humhub/components/bootstrap/ModuleAutoLoader.php

```diff
@@ -118,6 +119,12 @@ private static function findModules(iterable $paths): array
         return $modules;
     }
 
+    private static function getModuleConfigByPath(string $modulePath): array
+    {
+        return include $modulePath . DIRECTORY_SEPARATOR . self::CONFIGURATION_FILE;
```

Comment:
After running some tests for this addition I'm getting the following error;

```logs
TypeError: humhub\components\bootstrap\ModuleAutoLoader::getModuleConfigByPath(): Return value must be of type array, int returned in /home/LOCAL/protected/humhub/components/bootstrap/ModuleAutoLoader.php:124
Stack trace:
#0 /home/LOCAL/protected/humhub/components/bootstrap/ModuleAutoLoader.php(87): humhub\components\bootstrap\ModuleAutoLoader::getModuleConfigByPath()
#1 /home/LOCAL/protected/humhub/components/bootstrap/ModuleAutoLoader.php(55): humhub\components\bootstrap\ModuleAutoLoader::findModules()
#2 /home/LOCAL/protected/humhub/components/bootstrap/ModuleAutoLoader.php(39): humhub\components\bootstrap\ModuleAutoLoader::locateModules()
#3 /home/LOCAL/protected/vendor/yiisoft/yii2/base/Application.php(325): humhub\components\bootstrap\ModuleAutoLoader->bootstrap()
#4 /home/LOCAL/protected/vendor/yiisoft/yii2/web/Application.php(69): yii\base\Application->bootstrap()
#5 /home/LOCAL/protected/humhub/components/Application.php(61): yii\web\Application->bootstrap()
#6 /home/LOCAL/protected/vendor/yiisoft/yii2/base/Application.php(271): humhub\components\Application->bootstrap()
#7 /home/LOCAL/protected/humhub/components/Application.php(42): yii\base\Application->init()
#8 /home/LOCAL/protected/vendor/yiisoft/yii2/base/BaseObject.php(109): humhub\components\Application->init()
#9 /home/LOCAL/protected/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct()
#10 /home/LOCAL/protected/humhub/components/ApplicationTrait.php(38): yii\base\Application->__construct()
#11 /home/LOCAL/index.php(25): humhub\components\Application->__construct()
#12 {main}
```

To fix this and display better error handling here's a fix that will show exactly which `config.php` that causes any issues so that they can be fixed;
```php
    /**
     * Retrieves the module configuration from the specified path.
     *
     * @param string $modulePath The path to the module directory.
     * @return array The module configuration.
     * @throws \RuntimeException If the configuration file is not found or is invalid.
     */
    private static function getModuleConfigByPath(string $modulePath): array
    {
        $configFilePath = $modulePath . DIRECTORY_SEPARATOR . self::CONFIGURATION_FILE;

        // Check if the configuration file exists
        if (!file_exists($configFilePath)) {
            throw new \RuntimeException('Configuration file not found: ' . $configFilePath);
        }

        // Include the configuration file and ensure it returns an array
        $config = include $configFilePath;

        if (!is_array($config)) {
            throw new \RuntimeException('Invalid configuration file format: ' . $configFilePath);
        }

        return $config;
    }
```

If this method isn't wanted then I'd suggest only check `config.php` of enabled modules.

Reply:
To reproduce the error you must have a module that has a blank or missing `config.php` file.